core/posts/posts_view.py

```diff
@@ -0,0 +1,79 @@
+import base64
+import json
+
+from rest_framework.decorators import api_view
+from rest_framework.response import Response
+
+from core.posts.models import Posts
+from core.authors.models import Author
+from core.posts.serializers import PostsSerializer
+
+from core.authors.util import get_author_id
+
+def delete_post(request):
+    post_id = request.data["post_id"]
+    # Use post_id to delete all related image posts too
+    Posts.objects.filter(post_id=post_id).delete()
+    return True, "Post deleted"
+    
+
+def create_post(request):
+    success = True
+    message = "Post created"
+
+    body = request.data
+    data = json.loads(body["post_data"])
+
+    serializer = PostsSerializer(data=data)
+    serializer.is_valid()
+    new_post = serializer.save()
+
+    # if image, make another post with generated post_id
+    # TODO Multi Image Posts?
+    if (request.FILES):
+        new_id = new_post.post_id
+        image_type = request.FILES["imageFiles"].content_type
+        
+        if image_type == "image/jpeg" or image_type == "image/png":
+            content_type = f"{image_type};base64"
+        else:
```

Comment:
Shouldn't there be some kind of file size limit on the image?

Reply:
I would suggest enforcing a limit on both the frontend and backend.

Reply:
Added a 10MB backend limit